molecularnodes/handlers.py

```diff
@@ -46,20 +46,33 @@ def _selection_update_trajectories(self, context: bpy.types.Context) -> None:
 def update_trajectories(scene):
     "Call the set_frame method of all trajectories in the current session"
     session = scene.MNSession
+    print('Update trajectory has been called')
     for traj in session.trajectories.values():
         # use the updated method if it exists but otherwise fallback on the old method
         # of updating the trajectories
         if hasattr(traj, "update_with_scene"):
             if traj.update_with_scene:
                 traj.set_frame(scene.frame_current)
+                print("Hey change !")
             else:
                 traj.set_frame(traj.frame)
 
         else:
+            print("Hey change !")
             traj._update_positions(scene.frame_current)
             traj._update_selections()
             traj._update_calculations()
 
+
+@persistent
+def update_interactions(scene):
+    print("Update interactions has been called")
+    session = scene.MNSession
+    for entity in session.entities.values():
+        if entity._entity_type.value == "interaction":
+            print("An entity has been found")
+            entity.update_bond_positions(scene.frame_current)
+
```

Comment:
the `MolecularEntity` class has a `set_frame()` method. If you override it (with the contents of update_bond_positions) for the interaction class, then in the already existing `update_trajectories()` method above, can just loop through all entities, and call the `set_frame` method instead of creating a whole new per-frame update method. 

The plan is the make the `update_trajectories` function more generic, just calling it `update_entities` so that it can potentially update any kind of entity, but that will be done in another PR.

Reply:
I've put together the changes for making it more general in #721 , so once that is merged you can bring it into this branch and use the handler there. Sorry for the initial confusion around it, I thought I had already gone back and generalised it!